core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala

```diff
@@ -1103,7 +1103,7 @@ object ZSTMSpec extends ZIOBaseSpec {
           updater = ref.update(_ + 10).commit.forever
           res    <- (left <|> right).commit.race(updater)
```

Comment:
I think we need a way to yield to the runtime in the middle of evaluating `commit`. Right now we are racing an STM effect that retries as long as the reference is less than 500 with an effect that increments the reference. But it looks like since we are single threaded and we don't yield we are just caught on the left hand side and never complete. @mijicd Not sure what the best way to do this is. Any thoughts?

Reply:
I think `repeatN` and any other "spin loops" should yield in the middle. We can't really make `commit` yield without killing performance, and it shouldn't be necessary anyway if we don't have spin loops elsewhere.

Reply:
Imho, "early yield" approach you mentioned seems to be the only reasonable solution here.

Reply:
Agreed on the spin loops yielding in the middle but I don't think that helps here. In this example the right hand side is using `forever` and is yielding back to the runtime on every evaluation. But the right hand side is never getting evaluated at all because the left hand side is never completing or yielding because the transactional condition is never satisfied. In fact if I put a debug statement in `left` it looks like `left` is getting evaluated repeatedly in a busy loop, which I think is undesirable no matter what. So I think maybe if we fix that then if there are no more changes to the transactional variables the left hand side should be able to suspend which would then allow the runtime to use the thread to evaluate the right hand side and make progress?

Reply:
@mijicd Tried an alternative implementation of `orTry` that seems to avoid the issue. Would you take a look when you have the chance?